packages/runtime/src/widgets/Dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import type { Expression } from "framework";
+import { useRegisterBindings, useExecuteCode } from "framework";
+import { useNavigateScreen } from "../runtime/navigate";
+import { WidgetRegistry } from "../registry";
+import { MenuItem, Select, FormControl } from "@mui/material";
+import { SelectChangeEvent } from "@mui/material/Select";
+
+export type DropdownItem = {
+  label: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  id: Expression<string>;
+  label?: Expression<string>;
+  labelHint?: Expression<string>;
+  hintText?: Expression<string>;
+  required?: boolean;
+  enabled?: boolean;
+  autoComplete: boolean;
+  onChange?: {
+    executeCode?: string;
+    navigateScreen?: string;
+  };
+  styles?: {
+    minWidth?: number | string;
+    maxWidth?: number | string;
+    variant?: "standard" | "filled" | "outlined";
+    contentPadding: string | number;
+    fillColor: string;
+    borderRadius: number;
+    borderWidth: number;
+    borderColor: string;
+    disabledBorderColor: string;
+    errorBorderColor: string;
+    focusedBorderColor: string;
+    focusedErrorBorderColor: string;
+  };
+  defaultValue?: string;
+  menu?: (string | DropdownItem)[];
+};
+
+export const Dropdown: React.FC<DropdownProps> = (props) => {
+  const [dropdownValue, setDropdownValue] = React.useState(props.hintText);
+  const { values } = useRegisterBindings(
+    { ...props, dropdownValue },
+    props.id,
+    {
+      setDropdownValue,
+    },
+  );
+
+  const onTapCallback = useExecuteCode(
+    props.onChange?.executeCode || "",
+    values,
+  );
+  const onNavigate = useNavigateScreen(props.onChange?.navigateScreen || "");
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setDropdownValue(event.target.value);
+    props.onChange?.executeCode && onTapCallback();
+    props.onChange?.navigateScreen && onNavigate();
+  };
+
+  return (
+    <FormControl
```

Comment:
What is `FormControl` doing here? According to the docs, it provides context but only for a few components? 

https://mui.com/material-ui/api/form-control/#import